std/format.d

```diff
@@ -5693,3 +5933,56 @@ char[] sformat(Char, Args...)(char[] buf, in Char[] fmt, Args args)
 {
     return array1.ptr - array2.ptr;
 }
+
+@safe unittest
+{
+    assertCTFEable!({
+    auto tmp = format("%,d", 1000);
+    assert(tmp == "1,000", "'" ~ tmp ~ "'");
+
+    tmp = format("%,?d", 'z', 1234567);
+    assert(tmp == "1z234z567", "'" ~ tmp ~ "'");
+
+    tmp = format("%10,?d", 'z', 1234567);
+    assert(tmp == " 1z234z567", "'" ~ tmp ~ "'");
+
+    tmp = format("%11,2?d", 'z', 1234567);
+    assert(tmp == " 1z23z45z67", "'" ~ tmp ~ "'");
+
+    tmp = format("%11,*?d", 2, 'z', 1234567);
+    assert(tmp == " 1z23z45z67", "'" ~ tmp ~ "'");
+
+    tmp = format("%11,*d", 2, 1234567);
+    assert(tmp == " 1,23,45,67", "'" ~ tmp ~ "'");
+
+    tmp = format("%11,2d", 1234567);
+    assert(tmp == " 1,23,45,67", "'" ~ tmp ~ "'");
+    });
+}
+
+@safe unittest
+{
+    auto tmp = format("%,f", 1000.0);
+    assert(tmp == "1,000.000,000", "'" ~ tmp ~ "'");
+
+    tmp = format("%,f", 1234567.891011);
+    assert(tmp == "1,234,567.891,011", "'" ~ tmp ~ "'");
+
+    tmp = format("%,2f", 1234567.891011);
+    assert(tmp == "1,23,45,67.89,10,11", "'" ~ tmp ~ "'");
+
+    tmp = format("%18,f", 1234567.891011);
+    assert(tmp == " 1,234,567.891,011", "'" ~ tmp ~ "'");
+
+    tmp = format("%18,?f", '.', 1234567.891011);
+    assert(tmp == " 1.234.567.891.011", "'" ~ tmp ~ "'");
+
+    tmp = format("%.3,?f", 'ä', 1234567.891011);
+    assert(tmp == "1ä234ä567.891", "'" ~ tmp ~ "'");
+
+    tmp = format("%.3,*?f", 1, 'ä', 1234567.891011);
+    assert(tmp == "1ä2ä3ä4ä5ä6ä7.8ä9ä1", "'" ~ tmp ~ "'");
+
+    tmp = format("%.3,4?f", '_', 1234567.891011);
+    assert(tmp == "123_4567.891", "'" ~ tmp ~ "'");
```

Comment:
Hmm, should we standardize on `%,4.3f` instead of `%.3,4f`?  The full specifier looks better that way, e.g., `format("%12,3.3f", 1234.5678) == " 1,234.567"`.